sashimi/hardware/lasers/__init__.py

```diff
@@ -0,0 +1,54 @@
+from enum import Enum
+
+
+class LaserState(Enum):
+    ON = 1
+    OFF = 2
+
+
+class LaserException(Exception):
+    pass
+
+
+class LaserWarning(Warning):
+    pass
+
+
+class AbstractLaser:
+    def __init__(self, port):
+        self.port = port
+        self._status = LaserState.OFF
+
+    def start(self):
+        pass
+
+    def set_power(self, current):
+        """Sets power of laser based on either output power, applied current or power on sample"""
+        pass
+
+    def close(self):
+        pass
+
+    @property
+    def current(self):
```

Comment:
instead of current, I would call this `intensity` and have an additional attribute `intensity_units` to display in the GUI. Later we can think of adding transfer functions